### PR TITLE
Quaternion EulerAngles improvements and unit tests

### PR DESCRIPTION
**Pull request by gleblebedev**
EulerAngles should prioritize Yaw over Roll at singularity points as it fits 3D games better.
Unit tests to cover math changes.

### CODE REVIEW

File: Source/Tests/CSharp/Quaternion.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        [Fact]
+        public void QuaternionFromEuler()
+        {
+            Vector3 angles = new Vector3(10, -20, 30);
+            Matrix3 matrix = new Matrix3(angles.Y, new Vector3(0, 1, 0))
+                             * new Matrix3(angles.X, new Vector3(1, 0, 0))
+                             * new Matrix3(angles.Z, new Vector3(0, 0, 1));
+            Matrix3 expected = new Quaternion(angles).RotationMatrix;
+
+            matrix.ApproximatelyEquivalent(expected);
```
Comment (eugeneko) [inline Source/Tests/CSharp/Quaternion.cs:38]: Assert missing?

File: Source/Urho3D/CSharp/Managed/Math/Matrix3.cs [hunk 2]
```diff
@@ -62,6 +62,28 @@ public Matrix3(IReadOnlyList<float> data)
             M22 = data[8];
         }
 
+        /// <summary>
+        /// Construct from an angle (in degrees) and axis.
+        /// </summary>
+        public Matrix3(float angle, Vector3 axis)
```
Comment (eugeneko) [inline Source/Urho3D/CSharp/Managed/Math/Matrix3.cs:68]: I don't like that we now have two identical algorithms in different places.
`Quaternion(angle, axis).ToMatrix3()` and `Matrix3(angle, axis)` are copypaste basically. Any way we can get rid of it?

Comment (gleblebedev) [reply Source/Urho3D/CSharp/Managed/Math/Matrix3.cs:68]: There is no Quaternion.ToMatrix3() in c#

File: Source/Tests/CSharp/Quaternion.cs (hunk 1 above)
Comment (gleblebedev) [reply Source/Tests/CSharp/Quaternion.cs]: Fixed along with Equal() method unification.

File: Source/Urho3D/CSharp/Managed/Math/Matrix3.cs (hunk 2 above)
Comment (eugeneko) [reply Source/Urho3D/CSharp/Managed/Math/Matrix3.cs:70]: https://github.com/rbfx/rbfx/blob/master/Source/Urho3D/CSharp/Managed/Math/Quaternion.cs#L422
There totally is

Comment (gleblebedev) [reply Source/Urho3D/CSharp/Managed/Math/Matrix3.cs:70]: It isn't called ToMatrix3. Naming consistency isn't great :-(

Comment (eugeneko) [reply Source/Urho3D/CSharp/Managed/Math/Matrix3.cs:70]: I don't always remember exact function names. It's called the same way in C++ and in C# tho, so it's good enough

Comment (gleblebedev) [reply Source/Urho3D/CSharp/Managed/Math/Matrix3.cs:70]: Fixed

**Review by eugeneko — APPROVED** [review]

**Pull request merged**